### PR TITLE
Clean local development instructions

### PR DESCRIPTION
**Pull request by Sybrand**
**Summary:**
In the process of creating clean, easy to follow instructions for local development, I uncovered some hard coded variables that needed adjusting when not running web & pipeline containerised AND when running all servers (druid etc.) locally.

As part of this review, please review and follow instructions on the Wiki: https://github.com/Zenysis/Harmony/wiki/Local-Development-%E2%80%90-Step-by-step-instructions

- Introduced several variables for hard coded settings.
- Wrote [Step-by-step instructions on Wiki](https://github.com/Zenysis/Harmony/wiki/Local-Development-%E2%80%90-Step-by-step-instructions)
- Updated README to link to step by step instructions.
- Updated yarn.lock file (removed dependency)

**Test Plan:**
Run pipeline local and containerised - follow instructions from: https://github.com/Zenysis/Harmony/wiki/Local-Development-%E2%80%90-Step-by-step-instructions

### CODE REVIEW

**Review by tahitihat — CHANGES_REQUESTED** [review]
Wiki + README: I consider this an interim state — it's slightly confusing that the majority of the instructions live in the README but some information is available via GH Wiki. This is not a reason to avoid moving to the wiki, just flagging that docs should be consolidated in the future.

Testing: Following the local druid setup instructions, I can't run `make single_server_up` without volume mounting errors (on macOS, with DRUID_SHARED_FOLDER=/Users/isabel/home/share). I completely pruned docker in case my machine had conflicting volumes from other projects but no dice. I couldn't immediately find a problem in the compose files, so we'll have to check in tomorrow/later this week.

**Comment by Sybrand** [comment]
> Wiki + README: I consider this an interim state — it's slightly confusing that the majority of the instructions live in the README but some information is available via GH Wiki. This is not a reason to avoid moving to the wiki, just flagging that docs should be consolidated in the future.

Agreed - I think our options are carving off sections bit by bit (like this), or re-writing the whole README into the Wiki in one go. Right now we're carving...

> 
> Testing: Following the local druid setup instructions, I can't run `make single_server_up` without volume mounting errors (on macOS, with DRUID_SHARED_FOLDER=/Users/isabel/home/share). I completely pruned docker in case my machine had conflicting volumes from other projects but no dice. I couldn't immediately find a problem in the compose files, so we'll have to check in tomorrow/later this week.

Could you please give me some details? Or else schedule a call and we'll go over it.

File: db/druid/indexing/common.py [hunk 1]
```diff
@@ -75,10 +75,29 @@ def build_io_config_minio(
     }
 
 
+def fix_local_path(
```
[the remark this answers is not in the document]
Comment (Sybrand) [reply db/druid/indexing/common.py]: `_fix_path_for_druid` make more sense?

File: pipeline/template/index/run/00_druid/00_index [hunk 2]
```diff
@@ -2,6 +2,8 @@
 set -o pipefail
 
 "${PIPELINE_SRC_ROOT}/db/druid/indexing/scripts/run_native_indexing.py" \
-  --data_files='/home/share/data/SAMPLE/*/current/processed_rows.*' \
+  --data_files='${DRUID_SHARED_FOLDER:-/home/share}/data/SAMPLE/*/current/processed_rows.*' \
   --task_id_file="${PIPELINE_TMP_DIR}/task_id" \
+  --task_hash_dir "${DRUID_SHARED_FOLDER:-/home/share}/data/logs/druid_indexing/hash" \
+  --local_server_shared_folder "${DRUID_SHARED_FOLDER:-/home/share}" \
```
[the remark this answers is not in the document]
Comment (Sybrand) [reply pipeline/template/index/run/00_druid/00_index:8]: Sorry, not fully understanding - are you saying make `--task_hash_dir` required?

File: scripts/db/postgres/dev/create_default_admin_user.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ def main():
 
     first_name = 'Software'
     last_name = 'Developer'
-    email = 'software@something.com'
```
[the remark this answers is not in the document]
Comment (Sybrand) [reply scripts/db/postgres/dev/create_default_admin_user.py:21]: The poor people at software.com would start getting strange emails....

File: pipeline/template/index/run/00_druid/00_index (hunk 2 above)
[the remark this answers is not in the document]
Comment (tahitihat) [reply pipeline/template/index/run/00_druid/00_index:8]: Yes —

**Comment by tahitihat** [comment]
Okay, slacking you👍 I don't know exactly when I should schedule a call across our post Daylight Savings time difference. Will PM.

**Comment by Sybrand** [comment]
> 
> Testing: Following the local druid setup instructions, I can't run `make single_server_up` without volume mounting errors (on macOS, with DRUID_SHARED_FOLDER=/Users/isabel/home/share). I completely pruned docker in case my machine had conflicting volumes from other projects but no dice. I couldn't immediately find a problem in the compose files, so we'll have to check in tomorrow/later this week.

I've just removed the named volumes entirely - it causes too much headaches. Now, with a direct volume mapping, you can change without having to worry about a pesky volume hanging around.

**Comment by AbbyBeeler** [comment]
@zenenock can you take reviewing this?

**Comment by zenenock** [comment]
I'll review today

**Pull request merged**